### ISSUE TITLE
ColorPicker: Color space dropdown has no focus style

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#50524]
### Description

Focusing on the color space drop-down in the `ColorPicker` component does not change the style. Therefore, when the focus is moved by keyboard operation, it is not visible that this dropdown has the focus:

https://github.com/WordPress/gutenberg/assets/54422211/5867cb89-f8a8-4793-ab60-0a00e7933655

The reason is that [the BackdropUI to which the focus style is applied is hidden](https://github.com/WordPress/gutenberg/blob/1a06ca12c77b148d5730ffc4567ff18722035339/packages/components/src/color-picker/styles.ts#L33).

I would expect this element to be hidden for design reasons, but it should only be visible when it is in focus.

### Step-by-step reproduction instructions

N/A

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by afercia** [comment]
Looks like this regressed with the design refactoring in https://github.com/WordPress/gutenberg/pull/34598
